### DISCUSSION TITLE
Can an admin create /delete a Spark cluster for other users / groups?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#5047] [Q&A]
I want a Kyuubi administrator to create / delete Spark clusters for any other users/groups. How to do it?

**Comment by HaoYang670** [comment]
I have tried setting `"hive.server2.proxy.user": "user1"` in the the config when creating a session and providing the real username by authentication header, which I have set it as an administrator in `kyuubi.server.administrators`. 

But Kyuubi tells me ` "Method Not Allowed: Failed to validate proxy privilege of ... for user1",`

With the same proxy user and admin, I can successfully get a 200 response from `listEngine` request. So I don't know what wrong with the `open session`.

Hi @pan3793 Could you please take a look if you have time?

**Reply by HaoYang670** [reply]
Oh, I notice that we don't check if the `realUser` is admin or not in `getProxyUser`. Is it expected?

**Reply by pan3793** [reply]
cc @turboFei

**Reply by turboFei** [reply]
it is better to check if the real user is admin I think.

**Reply by HaoYang670** [reply]
Thanks. I will file an issue for this.

**Reply by HaoYang670** [reply]
Filed https://github.com/apache/kyuubi/issues/5050
